src/js/hyperschedule.js

```diff
@@ -536,16 +539,55 @@ function coursePassesTextFilters(course, textFilters) {
   const lowerCourseCode = course.courseCode.toLowerCase();
   const dept = lowerCourseCode.split(" ")[0];
   const col = lowerCourseCode.split(" ")[2].split("-")[0];
+  const credits = course.courseCredits;
 
   if (
     (textFilters["dept:"] && !dept.match(textFilters["dept:"])) ||
-    (textFilters["college:"] && !col.match(textFilters["college:"]))
+    (textFilters["college:"] && !col.match(textFilters["college:"])) ||
+    (textFilters["credits:"] &&
+      !courseCreditMatch(credits, textFilters["credits:"]))
   ) {
     return false;
   }
   return true;
 }
 
+function courseCreditMatch(courseCredits, inputCredits) {
```

Comment:
I suggest splitting out the parsing logic into a separate function, partly because I (stylistically) prefer seeing more `const`s than `let`s, and for efficiency reasons.  Below is some suggested code:

```js
function parseCreditsInequality(inputCredits) {
  for (const rel of filterInequalities)
    if (inputCredits.startsWith(rel)) 
      return rel;
  return "";
}

function parseCreditsFilter(inputCredits) {
  const rel = parseCreditsInequality(inputCredits);
  return { rel, floatInput: parseFloat(inputCredits.substring(rel.length) };
} 

function courseCreditMatch(courseCredits, inputCredits) {
  const { rel, floatInput } = parseCreditsFilter(inputCredits);
  const floatCredits = parseFloat(courseCredits);
```